turbinia/lib/recipe_helpers.py

```diff
@@ -0,0 +1,166 @@
+# -*- coding: utf-8 -*-
+# Copyright 2021 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Library to contain recipe validation logic."""
+
+import copy
+import logging
+import yaml
+from yaml import Loader
+from yaml import load
+from turbinia.lib.file_helpers import file_to_str
+from turbinia.lib.file_helpers import file_to_list
+from turbinia.client import TASK_MAP
+
+log = logging.getLogger('turbinia')
+
+#Attributes allowed on the 'globals' task recipe
+DEFAULT_GLOBALS_RECIPE = {
+    'debug_tasks': False,
+    'jobs_allowlist': [],
+    'jobs_denylist': [],
+    'yara_rules_file': None,
+    'filter_patterns_file': None,
+    'yara_rules': '',
+    'filter_patterns': ''
+}
+
+#Default recipes dict
+DEFAULT_RECIPE = {'globals': DEFAULT_GLOBALS_RECIPE}
+
+
+def load_recipe_from_file(recipe_file):
+  """Load recipe from file.
+
+  Args:
+    recipe_file(str): Name of the recipe file to be read.
+
+  Returns:
+    dict: Validated and corrected recipe dictionary.
+        Empty dict if recipe is invalid.
+  """
+  if not recipe_file:
+    return copy.deepcopy(DEFAULT_RECIPE)
+  try:
+    with open(recipe_file, 'r') as r_file:
```

Comment:
Maybe in a separate PR later we can have it check for the file name as passed in and the file name with '.yaml' appended.